File: app/src/main/java/com/botijasoftware/utils/animation/Skeleton.java
package com.botijasoftware.utils.animation;

import java.util.ArrayList;

public class Skeleton {

	public ArrayList<Animation> animations = new ArrayList<Animation>();
	public ArrayList<Bone> bones = new ArrayList<Bone>();
	public Bone rootBone;
	public Animation currentAnimation;
	
	public void Update(float time) {}

	public Animation findAnimation(String animationname) {
		for (int i=0;i<animations.size();i++) {
			Animation a = animations.get(i); 
			if (a.name.equalsIgnoreCase(animationname)) {
				return a;
			}
		}
		return null;
	}
	
	public Bone findBone(String bonename) {
		for (int i=0;i<bones.size();i++) {
			Bone b = bones.get(i); 
			if (b.name.equalsIgnoreCase(bonename)) {
				return b;
			}
		}
		return null;
	}
	
}

File: app/src/main/java/com/botijasoftware/utils/animation/Skeleton.kt
package com.botijasoftware.utils.animation

import java.util.ArrayList

class Skeleton {

    var animations = ArrayList<Animation>()
    var bones = ArrayList<Bone>()
    var rootBone: Bone? = null
    var currentAnimation: Animation? = null

    fun Update(time: Float) {}

    fun findAnimation(animationname: String): Animation? {
        for (i in animations.indices) {
            val a = animations[i]
            if (a.name.equals(animationname, ignoreCase = true)) {
                return a
            }
        }
        return null
    }

    fun findBone(bonename: String): Bone? {
        for (i in bones.indices) {
            val b = bones[i]
            if (b.name.equals(bonename, ignoreCase = true)) {
                return b
            }
        }
        return null
    }

}